File: 冯娆/python_demo/pages/login_page.py
from selenium.webdriver.common.by import By
from pages.news_page import MainPage
from pages.base_page import BasePage


class LoginPage(BasePage):

    _userphone_locator=(By.ID,"com.example.lenovo.enjoyball:id/et_login_phone")
    _password_locator = (By.ID, "com.example.lenovo.enjoyball:id/et_login_pwd")
    _btnsubmit_locator = (By.ID, "com.example.lenovo.enjoyball:id/bt_login_lg")
    _toast_locator=(By.XPATH,"//*[@class='android.widget.Toast']")

    def login(self,userphone,password):
        user_phone = self.find_element(self._userphone_locator)
        user_phone.send_keys(userphone)
        user_password = self.find_element(self._password_locator)
        user_password.send_keys(password)
        self.find_element(self._btnsubmit_locator).click()

    def login_success(self,userphone,password):
        self.login(userphone,password)
        return MainPage(self.driver)

    # def login_fail(self, userphone, password):
    #     self.login(userphone, password)
    #     return self

    def get_toast_txt(self):
        return self.find_element(self._toast_locator).text

    def get_current_activity(self):
        return self.driver.current_activity

File: java+appium自动化测试/enjoyball-appium/src/edu/pages/LoginPage.java
package edu.pages;

import edu.utils.BaseAction;
import io.appium.java_client.android.AndroidDriver;
import io.appium.java_client.android.AndroidElement;

public class LoginPage {
	private AndroidDriver<AndroidElement> driver;
	BaseAction action;

	public LoginPage(AndroidDriver<AndroidElement> driver) {
		this.driver = driver;
		this.action = new BaseAction(driver);
	}

	public AndroidElement edit_userphone() {
		return this.driver.findElementById("com.example.lenovo.enjoyball:id/et_login_phone");
	}

	public AndroidElement edit_password() {
		return this.driver.findElementById("com.example.lenovo.enjoyball:id/et_login_pwd");
	}

	public AndroidElement btn_submit() {
		return this.driver.findElementById("com.example.lenovo.enjoyball:id/bt_login_lg");
	}

	public HomePage loginSuccess(String name, String password) {

		action.type(edit_userphone(), name);
		action.type(edit_password(), password);
		action.click(btn_submit());

		return new HomePage(driver);
	}

	public String  loginFail(String name, String password) {

		action.type(edit_userphone(), name);
		action.type(edit_password(), password);
		action.click(btn_submit());
		return action.getToast();
	}
}
